src/main/java/com/yammer/dropwizard/authenticator/LdapAuthenticator.java

```diff
@@ -84,6 +87,14 @@ private boolean filterByGroup(InitialDirContext context, String sanitizedUsernam
         }
     }
 
+    private String userNameBaseOnGroupClass(String userName) {
+        if ("groupOfNames".equalsIgnoreCase(configuration.getGroupClassName())
+                && "member".equalsIgnoreCase(configuration.getGroupMembershipAttribute())) {
```

Comment:
"groupOfNames" and "member" seem pretty specific to your LDAP setup use case. Other setups can use other fields for their member identity :(